packages/loader/driver-utils/src/buildSnapshotTree.ts

```diff
@@ -14,26 +14,27 @@ import {
     TreeEntry,
 } from "@fluidframework/protocol-definitions";
 import { buildHierarchy } from "@fluidframework/protocol-base";
+import { v4 as uuid } from "uuid";
 
-async function flattenCore(
+function flattenCore(
     path: string,
     treeEntries: ITreeEntry[],
     blobMap: Map<string, string>,
-): Promise<git.ITreeEntry[]> {
+): git.ITreeEntry[] {
     const entries: git.ITreeEntry[] = [];
     for (const treeEntry of treeEntries) {
         const subPath = `${path}${treeEntry.path}`;
 
         if (treeEntry.type === TreeEntry.Blob) {
             const blob = treeEntry.value as IBlob;
             const buffer = IsoBuffer.from(blob.contents, blob.encoding);
-            const sha = await gitHashFile(buffer);
-            blobMap.set(sha, buffer.toString("base64"));
+            const id = uuid();
```

Comment:
This is the crux of the change. As far as i can tell this is safe, as this SHA is really only used as an Id for our internal representation of the snapshot and is never pushed back to the server

Reply:
I believe you are right! We can go even cheaper - how about static counter? :)